### PR TITLE
{CI} Add extension versioning doc in pr template

### PR DESCRIPTION
**Pull request by AllyW**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

**Comment by AllyW** [comment]
image

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -15,6 +15,9 @@ For new extensions:
 
 - [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).
```
Comment (wangzelin007) [inline .github/pull_request_template.md:16]: How about adding the `extension version` to the `For new extensions:` section?
`- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)`

Comment (AllyW) [reply .github/pull_request_template.md]: Cause all existing modules need to comply this rule in future releases, so put it in general guidelines.

**Comment by AllyW** [comment]
image

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**